From 96208f4f0225ecda345a882618cc6e4721b81230 Mon Sep 17 00:00:00 2001
From: Keyon Jie <yang.jie@linux.intel.com>
Date: Mon, 2 Apr 2018 19:00:44 +0800
Subject: [PATCH 1312/2278] From 16c017ee36429a76986674561955f1844250b632 Mon
 Sep 17 00:00:00 2001 From: Liam Girdwood <liam.r.girdwood@linux.intel.com>
 Date: Fri, 3 Nov 2017 20:56:42 +0000 Subject: [PATCH 45/70] ALSA: HACK: Fix
 rmmod crash

Something appears to corrupt the runtime->status on second playback...

Signed-off-by: Liam Girdwood <liam.r.girdwood@linux.intel.com>
Signed-off-by: Keyon Jie <yang.jie@linux.intel.com>
---
 sound/core/pcm.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/sound/core/pcm.c b/sound/core/pcm.c
index 2b5caa8..14ccc74f 100644
--- a/sound/core/pcm.c
+++ b/sound/core/pcm.c
@@ -1160,6 +1160,7 @@ static int snd_pcm_dev_disconnect(struct snd_device *device)
 	list_del_init(&pcm->list);
 	for (cidx = 0; cidx < 2; cidx++) {
 		for (substream = pcm->streams[cidx].substream; substream; substream = substream->next) {
+#if 0
 			snd_pcm_stream_lock_irq(substream);
 			if (substream->runtime) {
 				substream->runtime->status->state = SNDRV_PCM_STATE_DISCONNECTED;
@@ -1167,6 +1168,7 @@ static int snd_pcm_dev_disconnect(struct snd_device *device)
 				wake_up(&substream->runtime->tsleep);
 			}
 			snd_pcm_stream_unlock_irq(substream);
+#endif
 		}
 	}
 	if (!pcm->internal) {
-- 
2.7.4

